Application/Education/Certificate/Generator/Service/DataCertificate/IDataESBD.php

```diff
@@ -0,0 +1,969 @@
+                        $Data->createCertificateLevel($tblCertificate, $tblLevel);
+                    }
+                }
+            }
+////            // Begrenzung des Bemerkungsfelds
+//            $FieldName = 'Remark';
+//            if (!$Data->getCertificateFieldByCertificateAndField($tblCertificate, $FieldName)){
+//                $Data->createCertificateField($tblCertificate, $FieldName, 2300);
+//            }
+            // Begrenzung des Einschätzungfelds // Die Zeichenzählung lassen wir erstmal weg, da Schule Workaround über eine Wordvorlage hat.
+//            $FieldName = 'Rating';
+//            if (!$Data->getCertificateFieldByCertificateAndField($tblCertificate, $FieldName)){
+//                $Data->createCertificateField($tblCertificate, $FieldName, 2100);
+//            }
+        }
+        if ($tblCertificate && !$Data->getCertificateGradeAll($tblCertificate)) {
+            $Data->setCertificateGradeAllStandard($tblCertificate);
+        }
+        if ($tblCertificate && !$Data->getCertificateSubjectAll($tblCertificate)) {
+
+            $Data->setCertificateSubject($tblCertificate, 'DE', 1, 1);
+            $Data->setCertificateSubject($tblCertificate, 'SU', 1, 2);
+
+            $Data->setCertificateSubject($tblCertificate, 'MA', 2, 1);
+            $Data->setCertificateSubject($tblCertificate, 'EN', 2, 2);
+        }
+    }
+    /**
+     * @param Data        $Data
+     * @param TblConsumer $tblConsumerCertificate
+     */
+    private static function setEsbdGsJOne(Data $Data, TblConsumer $tblConsumerCertificate)
+    {
+
+        $tblCertificate = $Data->createCertificate('Grundschule Jahreszeugnis', 'Klasse 1-3', 'ESBD\EsbdGsJOne', $tblConsumerCertificate);
+        if ($tblCertificate) {
+            if ($Data->getTblSchoolTypePrimary()) {
+                $Data->updateCertificate($tblCertificate, $Data->getTblCertificateTypeYear(), $Data->getTblSchoolTypePrimary());
+                if (!$Data->getCertificateLevelAllByCertificate($tblCertificate)) {
+                    if (($tblLevel = Division::useService()->getLevelBy($Data->getTblSchoolTypePrimary(), '1'))) {
+                        $Data->createCertificateLevel($tblCertificate, $tblLevel);
+                    }
+                    if (($tblLevel = Division::useService()->getLevelBy($Data->getTblSchoolTypePrimary(), '2'))) {
+                        $Data->createCertificateLevel($tblCertificate, $tblLevel);
+                    }
+                    if (($tblLevel = Division::useService()->getLevelBy($Data->getTblSchoolTypePrimary(), '3'))) {
+                        $Data->createCertificateLevel($tblCertificate, $tblLevel);
+                    }
+                }
+            }
+        }
+    }
+    /**
+     * @param Data        $Data
+     * @param TblConsumer $tblConsumerCertificate
+     */
+    private static function setEsbdGymHjInfo(Data $Data, TblConsumer $tblConsumerCertificate)
```

Comment:
Codacy found an issue: [The method setEsbdGymHjInfo() has a Cyclomatic Complexity of 13. The configured cyclomatic complexity threshold is 10.](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5757982)